server/application/terminal.go

```diff
@@ -60,71 +58,84 @@ func (s *terminalHandler) getApplicationClusterRawConfig(ctx context.Context, a
+	if len(validationErrors) == 0 {
+		return name
+	}
+	return ""
 }
 
-// isValidNamespaceName checks that a namespace name is valid
-func isValidNamespaceName(name string) bool {
-	// https://github.com/kubernetes/kubernetes/blob/976a940f4a4e84fe814583848f97b9aafcdb083f/pkg/apis/core/validation/validation.go#L262
-	isValid := apimachineryvalidation.ValidateNamespaceName(name, false)
-	return len(isValid) == 0
+func getValidAppName(name string) string {
+	// app names have the same rules as pods.
+	return getValidPodName(name)
+}
+
+func getValidProjectName(name string) string {
+	// project names have the same rules as pods.
+	return getValidPodName(name)
 }
 
-// isValidContainerName checks that a containerName is valid
-func isValidContainerName(name string) bool {
-	// quick check to ensure we aren't passing along anything unsafe
-	isQualified := validation.IsQualifiedName(name)
-	return len(isQualified) == 0
+// getValidNamespaceName checks that a namespace name is valid
+func getValidNamespaceName(name string) string {
+	// https://github.com/kubernetes/kubernetes/blob/976a940f4a4e84fe814583848f97b9aafcdb083f/pkg/apis/core/validation/validation.go#L262
+	validationErrors := apimachineryvalidation.ValidateNamespaceName(name, false)
+	if len(validationErrors) == 0 {
+		return name
+	}
+	return ""
 }
 
-// GetQueryValue returns a value for a given url key
-// and strips newline to prevent go/log-injection
-func GetQueryValue(q url.Values, key string) string {
-	return strings.Replace(q.Get(key), "\n", "", -1)
+// getValidContainerName checks that a containerName is valid
+func getValidContainerName(name string) string {
+	// https://github.com/kubernetes/kubernetes/blob/53a9d106c4aabcd550cc32ae4e8004f32fb0ae7b/pkg/api/validation/validation.go#L280
+	validationErrors := apimachineryvalidation.NameIsDNSLabel(name, false)
+	if len(validationErrors) == 0 {
+		return name
+	}
+	return ""
 }
 
 func (s *terminalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	q := r.URL.Query()
-	podName := GetQueryValue(q, "pod")
-	container := GetQueryValue(q, "container")
-	app := GetQueryValue(q, "appName")
-	namespace := GetQueryValue(q, "namespace")
-	shell := GetQueryValue(q, "shell")
 
-	if podName == "" || container == "" || app == "" || namespace == "" {
+	if q.Get("pod") == "" || q.Get("container") == "" || q.Get("app") == "" || q.Get("project") == "" || q.Get("namespace") == "" {
 		http.Error(w, "Missing required parameters", http.StatusBadRequest)
 		return
 	}
 
-	// validate query string parameters to prevent unsafe usage
-	if !isValidNamespaceName(namespace) {
-		http.Error(w, "Namespace name is not valid", http.StatusBadRequest)
-		return
-	}
-	if !isValidPodName(podName) {
+	podName := getValidPodName(q.Get("pod"))
+	if podName == "" {
 		http.Error(w, "Pod name is not valid", http.StatusBadRequest)
 		return
 	}
-	if !isValidContainerName(container) {
+	container := getValidContainerName(q.Get("container"))
+	if podName == "" {
```

Comment:
I think you want to check `container` variable not `podName`

Reply:
don't code while sick, folks. :-P